crates/wit-parser/src/ast.rs

```diff
@@ -545,6 +558,48 @@ enum InterfaceItem<'a> {
     TypeDef(TypeDef<'a>),
     Func(NamedFunc<'a>),
     Use(Use<'a>),
+    Nest(Nest<'a>),
+}
+
+struct Nest<'a> {
+    docs: Docs<'a>,
+    id: PackageName<'a>,
+    name: Id<'a>,
+    attributes: Vec<Attribute<'a>>,
+}
+
+impl<'a> Nest<'a> {
+    fn parse(
+        tokens: &mut Tokenizer<'a>,
+        docs: Docs<'a>,
+        attributes: Vec<Attribute<'a>>,
+    ) -> Result<Self> {
+        tokens.eat(Token::Nest)?;
+        let id = parse_id(tokens)?;
+        tokens.expect(Token::Colon)?;
+        // `foo:bar/baz@1.0`
+        let namespace = id;
+        let pkg_name = parse_id(tokens)?;
+        tokens.expect(Token::Slash)?;
+        let name = parse_id(tokens)?;
+        let version = parse_opt_version(tokens)?;
```

Comment:
I think that this is a duplication of `UseName`'s internal parsing, but could `UseName` be used here instead? That would enable consistently being able to refer to both foreign and local interfaces with `nest`